Bowling/Bowling/Score.cs

```diff
@@ -0,0 +1,167 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class Score
+    {
+        private int _frameNumberTotal;
+
+        public Score()
+        {
+            _frameNumberTotal = 10;
+        }
+
+        public List<Frame> CreateScoreCard(string bowlingMarks)
+        {
+            var frameNumberWithBonusRoll = 10;
+            var frameNumber = 1;
+            List<Frame> framesCollection = new List<Frame>();
+            ArrayList bowlingMarksByFrame = new ArrayList(OrganizeBowlingMarksByFrame(bowlingMarks));
+
+            foreach (var marks in bowlingMarksByFrame)
+            {
+                Frame frame = new Frame();
+
+                if (frameNumber != frameNumberWithBonusRoll)
+                {
+                    FrameNoBonusRollFactory frameNoBonusRoll = new FrameNoBonusRollFactory();
+                    frame = frameNoBonusRoll.Create(frameNumber, marks.ToString());
+                }
+                else
+                {
+                    FrameWithBonusRollFactory frameWithBonusRoll = new FrameWithBonusRollFactory();
+                    frame = frameWithBonusRoll.Create(marks.ToString());
+                }
+
+                framesCollection.Add(frame);
+                frameNumber++;
+            }
+            
+            return framesCollection;
+        }
+
+        public int CalculateScoreCardTotals(List<Frame> framesCollection)
+        {
+            List<Frame> updatedFramesCollection = UpdateFrameScoresForStrikesAndSpares(framesCollection);
+            var grandTotal = CalculateFramesGrandTotal(updatedFramesCollection);
+
+            return grandTotal;
+        }
+
+        private ArrayList OrganizeBowlingMarksByFrame(string bowlingMarks)
+        {
+            var frameMarks = String.Empty;
+            var bowlMark = String.Empty;
+            var frameCount = 1;
+            ArrayList bowlingMarksByFrame = new ArrayList();
+
+            foreach(var mark in bowlingMarks)
+            {
+                bowlMark = mark.ToString();
+                frameMarks += bowlMark;
+
+                if (frameCount != _frameNumberTotal)
+                {
+                    if (bowlMark.Equals(RollMarks.strikeMark))
+                    {
+                        bowlingMarksByFrame.Add(frameMarks);
+                        frameMarks = String.Empty;
+                        frameCount++;
+                    }
+                    else if (frameMarks.Length == 2)
+                    {
+                        bowlingMarksByFrame.Add(frameMarks);
+                        frameMarks = String.Empty;
+                        frameCount++;
+                    }
+                }
+            }
+
+            bowlingMarksByFrame.Add(frameMarks);
+            return bowlingMarksByFrame;
+        }
+
+
+        private List<Frame> UpdateFrameScoresForStrikesAndSpares(List<Frame> framesCollection)
```

Comment:
Given my feedback about ternary operators, `IEnumerable`, `yield`, etc...try to apply those tools to this function to see how it can be simplified.

This may be a good opportunity for us to pair and I can demonstrate how minalmalistic we can make this!